### PR TITLE
[rel/5.x.x] Implement Cobertura coverage format

### PR DESCRIPTION
**Pull request by nohwnd**
Backport #2298 to 5.x.x
Backport #2553 as well

### CODE REVIEW

**Comment by nohwnd** [comment]
regarding your comments on the source PR @fflaten why do you say this one will need to use a different implementation? the api seems to exist in .NET Framework 462: https://learn.microsoft.com/en-us/dotnet/api/system.datetimeoffset.tounixtimemilliseconds?view=netframework-4.6.2 which is the oldest one we support.

the comments: 
https://github.com/pester/Pester/pull/2298#discussion_r1676502351
https://github.com/pester/Pester/pull/2298#discussion_r1676499300

**Comment by nohwnd** [comment]
Too tired to deal with the build errors right now.

**Comment by fflaten** [comment]
> regarding your comments on the source PR @fflaten why do you say this one will need to use a different implementation? the api seems to exist in .NET Framework 462

Pester v5 targets .NET 4.5.2.

File: tst/functions/Coverage.Tests.ps1 [hunk 1]
```diff
@@ -79,11 +86,18 @@ InPesterModuleScope {
...
+                        'I am the constructor of base class.'
+                    }
+                #}
+
                 #class MyClass
                 #{
                     function MyClass
                     {
-                        'I am the constructor.'
+                        MyBaseClass # calls the base constructor
```
Comment (nohwnd) [inline tst/functions/Coverage.Tests.ps1:100]: This was originally `MyBaseClass; "I am the constructor"` but that adds one more BP than needed, removed the string so we simulate the call to base constructor like the new class base code does it.

Comment (nohwnd) [reply tst/functions/Coverage.Tests.ps1:100]: https://github.com/pester/Pester/pull/2577 Forward ported here.

**Pull request merged**